codalab/apps/teams/views.py

```diff
@@ -0,0 +1,324 @@
+                            if req.is_active and req.is_request and req != membership:
+                                req.status = TeamMembershipStatus.objects.get(codename=TeamMembershipStatus.CANCELED)
+                                req.end_date = now()
+                                req.save()
+
+        context = self.get_context_data()
+
+        return super(TeamDetailView, self).render_to_response(context)
+
+    def get_context_data(self, **kwargs):
+        context = super(TeamDetailView, self).get_context_data(**kwargs)
+        context['team'] = None
+        competition = Competition.objects.get(pk=self.kwargs['competition_pk'])
+        context['competition'] = competition
+
+        members_columns = [
+            {
+                'label': '#',
+                'name': 'number'
+            },
+            {
+                'label': 'NAME',
+                'name': 'name'
+            },
+            {
+                'label': 'EMAIL',
+                'name': 'email'
+            },
+            {
+                'label': 'JOINED',
+                'name' : 'joined'
+            },
+            {
+                'label': 'STATUS',
+                'name': 'status'
+            },
+            {
+                'label': 'ENTRIES',
+                'name': 'entries'
+            }
+        ]
+
+        if competition.participants.filter(user__in=[self.request.user]).exists():
+            participant = competition.participants.get(user=self.request.user)
+            if participant.status.codename == ParticipantStatus.APPROVED:
+                user_team = get_user_team(participant, competition)
+
+                team_list= get_competition_teams(competition)
+                user_requests = get_user_requests(participant, competition)
+                user_pending_teams = get_competition_user_pending_teams(competition, participant)
+                if user_team is not None:
+                    context['team'] = user_team
+                    context['team_requests'] = get_team_pending_membership(user_team)
+                    owner_part=competition.participants.get(user=user_team.creator)
+                    member_list=[
+                        {
+                            'pk': user_team.creator.pk,
+                            'name': user_team.creator.username,
+                            'email': user_team.creator.email,
+                            'joined': user_team.created_at,
+                            'status': 'creator',
+                            'number':  1,
+                            'entries': len(CompetitionSubmission.objects.filter(team=user_team, participant=owner_part)),
+                        }
+                    ]
+                    s_ap = TeamMembershipStatus.objects.get(codename=TeamMembershipStatus.APPROVED)
+                    for number, member in enumerate(user_team.members.all()):
+                        membership_set = member.teammembership_set.filter(team=user_team)
```

Comment:
these nested queries, do they need to be nested? does this slow down significantly?